transport/tlscommon/tls_fips_test.go

```diff
@@ -30,6 +32,10 @@ import (
 // TestFIPSCertifacteAndKeys tests that encrypted private keys fail in FIPS mode
 func TestFIPSCertificateAndKeys(t *testing.T) {
 	t.Run("embed encrypted PKCS#1 key", func(t *testing.T) {
+		if checkGODEBUG() {
+			t.Skip("GODEBUG=fips140=only detected, avoiding panics")
```

Comment:
I still don't quite get why we should skip this. I would expect a `requirefips` tagged build should not get panics. Is there an issue somewhere that we should fix in order for this not to panic and I am not seeing? 🙂

Reply:
`GODEBUG=fips140=only` will panic if any non-FIPS approved algorithm is used.
The purpose of adding the buildkite step with this setting is to add automation so that we do not accidentally add things that break FIPS compliance (unit tests that just test `requirefips` are already running elsewhere)

Reply:
To be more clear; in our regular FIPS tests will test that `_, err = LoadTLSConfig(cfg)` returns an error (expected behaviour when we compile for FIPS).

When `GODEBUG=fips140=only` is used the test fails in setup as  `makeKeyCertPair()` panics